### DISCUSSION TITLE
import data trouble

### DISCUSSION THREAD

**Discussion by rlargett** [directus/directus#5863] [Archive]
I set up a quick module to import a csv file to populate a collection but it just gives me empty entries in my specified collection.

Im using a standard html <input type="file"> element that calls this method when clicked: 

```js
importCSV() {
	const fileInput = document.querySelector('input.csvFileInput')
	const formData = new FormData()
	formData.append('title', 'testing')
	formData.append('file', new Blob([fileInput.files[0]], {type: "text/csv"}))
	this.system.api.post('/utils/import/teams', formData)
		.then((res) => console.log('import done', res))
		.catch((err) => console.log('import error', err))
}
```

note: I was initially having trouble with the import where the Data Type for CSV wasnt getting set properly, it kept making it octet-stream. but I was able to resolve that by creating the Blob with the type set to text/csv.

Thanks!

**Comment by rijkvanzanten** [comment]
Hi Ron!


What's the structure of the CSV? Does it map to the columns in the collection? Does the CSV file start with the fields as headers?

**Reply by rijkvanzanten** [reply]
Also, can you try it without `new Blob` around the file? Just `formData.append('file', fileInput.files[0]);` should do the trick 🤔

**Reply by rlargett** [reply]
Hey Rijk,
Thanks for responding!

Here is the csv im testing
```
"Team Code","Name","Conference Code"
5,"Akron",875
8,"Alabama",911
9,"UAB",24312
27,"Appalachian State",818
28,"Arizona State",905
```
Yes I tried it without the blob first but it wasn't recognizing it as a csv file. I kept getting an error, "Can't import files of type application/octet-stream" or something like that

**Reply by rlargett** [reply]
Do I need to setup my collection/fields a certain way?

**Reply by rijkvanzanten** [reply]
Do those header names match the field keys in Directus? 

`"Team Code","Name","Conference Code"`

Should that be `team_code`, `name`, `conference_code` for example?

**Reply by rlargett** [reply]
AHH!! That was it! Thanks!

That's kind of a bummer because these csv files are provided to me from my client. they purchase them from a service. So I guess I have to change all the headers to be snake_case? bad csv formatting.

**Reply by rlargett** [reply]
Is there something I could use in the module to automatically convert the headings to snake_case before importing?

**Reply by rijkvanzanten** [reply]
There's currently no way to "map" input fields to fields in Directus, so you'll have to do some touch ups to map the headers correctly 👍🏻

**Reply by rlargett** [reply]
Ok, sounds good!! Thanks again for your help!!

**Comment by phamalda** [comment]
Good day, all. I know I'm coming into this conversation quite late, but I was hoping someone might lend a hand. I have files generated by a third-party source that has column names that contain spaces. Names like "Notice ID" and "Current Response Date." The corresponding attributes in my Directus collection are Notice_ID and Current_Response_Date. 
I am trying to set up a flow that will automatically import the data therein when a csv file lands in the applicable folder. If there no ways to map those headers to collection fields, do you have any advice on how to successfully pull that data in? I'd really appreciate it.

Oh! These incoming files also have date values in varying formats (Jun 30, 2025 12:00 PM EDT or 6/23/2025). I could use a little help in converting those to the proper OSI format to be successfully consumed. Thanks a ton.
